swift/aperture/main.swift

```diff
@@ -31,21 +19,35 @@ func record() {
   var coordinates = [String]()
   if cropArea != "none" {
     coordinates = CommandLine.arguments[3].components(separatedBy: ":")
-    if coordinates.count - 1 != 3 { // Number of ':' in the string
-      print("The coordinates for the crop rect must be in the format 'originX:originY:width:height'")
-      exit(2)
-    }
   }
 
-  recorder = Recorder(
-    destinationPath: destinationPath,
-    fps: fps,
-    coordinates: coordinates,
-    showCursor: showCursor,
-    highlightClicks: highlightClicks,
-    displayId: displayId!,
-    audioDeviceId: audioDeviceId
-  )
+  do {
+    recorder = try Recorder(
+      destinationPath: destinationPath,
+      fps: fps,
+      coordinates: coordinates,
+      showCursor: showCursor,
+      highlightClicks: highlightClicks,
+      displayId: displayId!,
+      audioDeviceId: audioDeviceId
+    )
+  } catch {
+    printErr(error)
+    exit(1)
+  }
+
+  recorder.onStart = {
+    print("R")
+  }
+
+  recorder.onFinish = {
+    exit(0)
+  }
+
+  recorder.onError = {
+    printErr($0)
+    exit(1)
+  }
```

Comment:
@LarsJK What do you think about this pattern for events? I realize delegates are more common in Cocoa, but it feels very magic and verbose. I much prefer this way. I considered using NSNotifications, but it's also very verbose.